ethcore/src/block_queue.rs

```diff
@@ -285,18 +285,24 @@ impl BlockQueue {
 	}
 
 	/// Mark given block and all its children as bad. Stops verification.
-	pub fn mark_as_bad(&mut self, hash: &H256) {
+	pub fn mark_as_bad(&mut self, hashes: &[H256]) {
 		let mut verification_lock = self.verification.lock().unwrap();
+		let mut processing = self.processing.write().unwrap();
```

Comment:
Before verification lock was obtained a the beginning, but processing write-lock was obtained inside the loop - I've just extracted that above the loop, maintaining the order.

Had a quick look on all functions and it's not clear in which order to lock verification and processing - I will look into this.